cfme/tests/candu/test_host_graph.py

```diff
@@ -81,3 +92,86 @@ def refresh():
         for data in graph.data_for_legends(leg).values():
             graph_data += float(data[leg].replace(',', '').replace('%', '').split()[0])
     assert graph_data > 0
+
+
+def compare_data(tb_data, gp_data, legends, tolerance=1):
+    """ Compare Utilization graph and table data.
+
+    Args:
+        tb_data : Data from Utilization table
+        gp_data : Data from Utilization graph
+        legends : Legends in graph; which will help for comparison
+        tolerance : Its error which we have to allow while comparison
+    """
+    for row in tb_data:
+        for key, data in gp_data.items():
+            if any([re.match(key, item) for item in row['Date/Time'].split()]):
+                for leg in legends:
+                    tb = row[leg].replace(',', '').replace('%', '').split()
+                    if tb:
+                        tb = round(float(tb[0]), 1)
+                        gp = round(
+                            float(data[leg].replace(',', '').replace('%', '').split()[0]), 1)
+                        assert abs(tb - gp) <= tolerance
+                    else:
+                        logger.warning("No {leg} data captured for DateTime: {dt}".format(
+                            leg=leg, dt=row['Date/Time']))
+
+
+@pytest.mark.uncollectif(lambda provider, interval:
+                         provider.one_of(RHEVMProvider) and
+                         interval == "Daily")
+@pytest.mark.parametrize('interval', INTERVAL)
+@pytest.mark.parametrize('graph_type', HOST_GRAPHS)
+def test_graph_screen(provider, interval, graph_type, host, enable_candu):
+    """Test Host graphs for hourly and Daily
+
+    prerequisites:
+        * C&U enabled appliance
+
+    Steps:
+        * Navigate to Host Utilization Page
+        * Check graph displayed or not
+        * Select interval(Hourly or Daily)
+        * Zoom graph to get Table
+        * Compare table and graph data
+    """
+    wait_for(
+        host.capture_historical_data,
+        delay=20,
+        timeout=1000,
+        message="wait for capturing host historical data")
+    host.wait_candu_data_available(timeout=1200)
+
+    view = navigate_to(host, 'candu')
+    view.options.interval.fill(interval)
+
+    # Check garph displayed or not
+    graph = getattr(view.interval_type, graph_type)
+    assert graph.is_displayed
+
+    def refresh():
+        provider.browser.refresh()
+        view.options.interval.fill(interval)
+
+    # wait, some time graph take time to load
+    wait_for(lambda: len(graph.all_legends) > 0,
+             delay=5, timeout=200, fail_func=refresh)
+
+    # zoom in button not available with normal graph in Host Utilization page.
+    # We have to use vm average graph for zoom in operation.
+    vm_avg_graph = getattr(view.interval_type, "{}_vm_avg".format(graph_type))
```

Comment:
@rlbabyuk: I would like to go with `AttributeError`. It will help us to debug things. If, I go with some `default` attribute. It will continue and maybe pass the test as it will select a default graph. hmm, If you agree better to add exception and logging error but, I don't think it's needed.